examples/nodejs/get-set-batch-perf-test/perf-test.ts

```diff
@@ -0,0 +1,239 @@
+    await this.runAsyncGetRequests(momento);
+
+    this.logger.info('Starting set batch requests');
+    await this.runSetBatchTests(momento);
+
+    this.logger.info('Starting get batch requests');
+    await this.runGetBatchTests(momento);
+
+    // flush the cache
+    await flushCache(momento, this.cacheName, this.logger);
+  }
+
+  async runAsyncSetRequests(momento: CacheClient): Promise<void> {
+    for (const setConfig of this.testConfiguration.sets) {
+      const context = initiatePerfTestContext();
+      while (getElapsedMillis(context.startTime) < this.testConfiguration.minimumRunDurationSecondsForTests * 1000) {
+        await this.sendAsyncSetRequests(momento, context, setConfig);
+      }
+      calculateSummary(context, setConfig.batchSize, setConfig.itemSizeBytes, RequestType.ASYNC_SETS, this.logger);
+    }
+  }
+
+  async runAsyncGetRequests(momento: CacheClient): Promise<void> {
+    for (const getConfig of this.testConfiguration.gets) {
+      // ensure that the cache is populated with the keys
+      await this.ensureCacheIsPopulated(momento, getConfig);
+      const context = initiatePerfTestContext();
+      while (getElapsedMillis(context.startTime) < this.testConfiguration.minimumRunDurationSecondsForTests * 1000) {
+        await this.sendAsyncGetRequests(momento, context, getConfig);
+      }
+      calculateSummary(context, getConfig.batchSize, getConfig.itemSizeBytes, RequestType.ASYNC_GETS, this.logger);
+    }
+  }
+
+  async runSetBatchTests(momento: CacheClient): Promise<void> {
+    for (const setConfig of this.testConfiguration.sets) {
+      const context = initiatePerfTestContext();
+      while (getElapsedMillis(context.startTime) < this.testConfiguration.minimumRunDurationSecondsForTests * 1000) {
+        await this.sendSetBatchRequests(momento, context, setConfig);
+      }
+      calculateSummary(context, setConfig.batchSize, setConfig.itemSizeBytes, RequestType.SET_BATCH, this.logger);
+    }
+  }
+
+  async runGetBatchTests(momento: CacheClient): Promise<void> {
+    for (const getConfig of this.testConfiguration.gets) {
+      // ensure that the cache is populated with the keys
+      await this.ensureCacheIsPopulated(momento, getConfig);
+      const context = initiatePerfTestContext();
+      while (getElapsedMillis(context.startTime) < this.testConfiguration.minimumRunDurationSecondsForTests * 1000) {
+        await this.sendGetBatchRequests(momento, context, getConfig);
+      }
+      calculateSummary(context, getConfig.batchSize, getConfig.itemSizeBytes, RequestType.GET_BATCH, this.logger);
+    }
+  }
+
+  private async sendAsyncSetRequests(
+    momento: CacheClient,
+    context: PerfTestContext,
+    setConfig: GetSetConfig
+  ): Promise<void> {
+    const setPromises: Promise<CacheSet.Response>[] = [];
+    for (let i = 0; i < setConfig.batchSize; i++) {
+      const key = `key-${i}`;
+      const value = 'x'.repeat(setConfig.itemSizeBytes);
```

Comment:
supernit; for large items this operation is non-trivial so you could do it outside of the loop.